purchase_request_group_by_date/models/stock_rule.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2022 Camptocamp
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+
+from odoo import api, models
+
+
+class StockRule(models.Model):
+    _inherit = "stock.rule"
+
+    # override to change returned domain based on company settings
+    @api.model
+    def _make_pr_get_domain(self, values):
+        domain = super()._make_pr_get_domain(values)
+        lst_domain = [dom for dom in domain]
+        lst_domain.pop(0)
+        lst_domain += [
+            ("state", "in", ("draft", "to_approve", "approved")),
+        ]
+        domain = tuple(lst_domain)
+        return domain
+
+    # override method from purchase_request to change existing pr handling
+    def create_purchase_request(self, procurement_group):
+        procurement = procurement_group[0]
+        rule = procurement_group[1]
+        purchase_request_model = self.env["purchase.request"]
+        purchase_request_line_model = self.env["purchase.request.line"]
+        cache = {}
+        pr = self.env["purchase.request"]
+        domain = rule._make_pr_get_domain(procurement.values)
+        if domain in cache:
+            pr = cache[domain]
+        elif domain:
+            pr = (
+                self.env["purchase.request"]
+                .search([dom for dom in domain])
+                .filtered(
+                    lambda x: procurement.product_id in x.line_ids.mapped("product_id")
```

Comment:
Is it not possible to search on `purchase.request.line` and avoid doing a search with a filter ?